pkg/smb/controllerserver.go

```diff
@@ -311,15 +311,17 @@ func (d *Driver) smbVolToCSI(vol *smbVolume, parameters map[string]string) *csi.
 }
 
 // Given a CSI volume id, return a smbVolume
-func (d *Driver) getSmbVolFromID(id string) (*smbVolume, error) {
-	volRegex := regexp.MustCompile("^([^/]+)/([^/]+)$")
+// a sample volume Id: smb-server.default.svc.cluster.local/share/pvc-4729891a-f57e-4982-9c60-e9884af1be2f
+func getSmbVolFromID(id string) (*smbVolume, error) {
+	volRegex := regexp.MustCompile("^([^/]+)/(.*)/([^/]+)$")
 	tokens := volRegex.FindStringSubmatch(id)
 	if tokens == nil {
 		return nil, fmt.Errorf("Could not split %q into server, baseDir and subDir", id)
 	}
+	source := "\\" + tokens[1] + "/" + tokens[2]
```

Comment:
should be `"//" + tokens[1] + "/" + tokens[2]` ?

Reply:
smb server `source` start with `\\`.

Reply:
no, for linux, it's
```
parameters:
  source: "//smb-server.default.svc.cluster.local/share"
```

and I have added a source path convert fix, even it's `//smb-server.default.svc.cluster.local/share`, would convert to `\\smb-server.default.svc.cluster.local\share` on Windows

Reply:
done